pkg/controller/datavolume-controller.go

```diff
@@ -497,12 +497,19 @@ func (r *DatavolumeReconciler) reconcileClone(log logr.Logger,
 
 	if !prePopulated && !pvcPopulated {
 		if pvc == nil {
+			storageClass, err := GetStorageClassByName(r.client, pvcSpec.StorageClassName)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			if storageClass == nil {
```

Comment:
So previously, in the case of host-assisted clone, it would go forward and create a PVC -> line 530 (even without a default storageclass, and without storage class defined), now it will return here with error. 

Is this the way we want to handle this case?

Reply:
Yeah definitely want to avoid logical changes. I think I took this too far, trying to reuse the storage class that gets passed around.

I'll have another go with more direct approach